applications/ray/kuberay-tpu-webhook/webhook_main_test.go

```diff
@@ -0,0 +1,595 @@
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name: "ray-head",
+							},
+						},
+					},
+				},
+			},
+			WorkerGroupSpecs: []rayv1.WorkerGroupSpec{
+				{
+					Replicas:    pointer.Int32(1),
+					MinReplicas: pointer.Int32(0),
+					MaxReplicas: pointer.Int32(10000),
+					NumOfHosts:  1,
+					GroupName:   groupNameStr,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name: "ray-worker",
+									Env: []corev1.EnvVar{
+										{
+											Name: "MY_POD_IP",
+											ValueFrom: &corev1.EnvVarSource{
+												FieldRef: &corev1.ObjectFieldSelector{
+													FieldPath: "status.podIP",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// RayCluster with 2x2x4 TPU topology worker group
+	testRayClusterMultiHostTPU = &rayv1.RayCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      instanceName,
+			Namespace: namespaceStr,
+		},
+		Spec: rayv1.RayClusterSpec{
+			HeadGroupSpec: rayv1.HeadGroupSpec{
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name: "ray-head",
+							},
+						},
+					},
+				},
+			},
+			WorkerGroupSpecs: []rayv1.WorkerGroupSpec{
+				{
+					Replicas:    pointer.Int32(1),
+					MinReplicas: pointer.Int32(0),
+					MaxReplicas: pointer.Int32(10000),
+					NumOfHosts:  4,
+					GroupName:   groupNameStr,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name: "ray-worker",
+									Env: []corev1.EnvVar{
+										{
+											Name: "MY_POD_IP",
+											ValueFrom: &corev1.EnvVarSource{
+												FieldRef: &corev1.ObjectFieldSelector{
+													FieldPath: "status.podIP",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func Test_GetReplicaIndex(t *testing.T) {
+	setupTest(t)
+
+	tests := map[string]struct {
+		sliceToWorkers     map[slice][]worker
+		numOfHosts         int32
+		numReplicas        int
+		additionalGroupStr string
+		numOfHosts2        int32
+		numReplicas2       int
+	}{
+		"single-host, single-slice worker group": {
+			// single-slice, replicaIndex should always be 0
+			numOfHosts:  1,
+			numReplicas: 1,
+		},
+		"single-host, multi-slice worker group": {
+			// multi-slice, replicaIndex should always be 0-numReplicas
+			numOfHosts:  1,
+			numReplicas: 4,
+		},
+		"multi-host, single-slice worker group": {
+			// single-slice, replicaIndex should always be 0
+			numOfHosts:  4,
+			numReplicas: 1,
+		},
+		"multi-host, multi-slice worker group": {
+			// multi-slice, replicaIndex should always be 0-numReplicas for 0-numOfHosts pods
+			numOfHosts:  4,
+			numReplicas: 4,
+		},
+		"multiple worker groups": {
```

Comment:
Do we need a test for deleting pods here?

Reply:
Added a test case to validate that a re-created pod will be assigned to the lowest index replica with created Pods < NumOfHosts.